protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -1,3 +1,28 @@
+//! # Decoder
+//!
+//! Provides utilities for decoding messages held by Sv2 frames, with or without Noise protocol
+//! support.
+//!
+//! It includes primitives to both decode encoded standard Sv2 frames and to decrypt
+//! and decode Noise-encrypted encoded Sv2 frames, ensuring secure communication when required.
+//!
+//! ## Usage
+//! All messages passed between Sv2 roles are encoded as Sv2 frames. These frames are decoded using
+//! primitives in this module. There are two types of decoders for reading these frames: one for
+//! regular Sv2 frames (`StandardDecoder`), and another for Noise-encrypted frames
+//! (`StandardNoiseDecoder`). Both decoders manage the deserialization of incoming data and, when
+//! applicable, the decryption of the data upon receiving the transmitted message.
+//!
+//! ### Buffer Management
+//!
+//! The decoders rely on buffers to hold intermediate data during the decoding process.
+//!
+//! - When the `with_buffer_pool` feature is enabled, the internal `Buffer` type is backed by a
+//!   pool-allocated buffer (`binary_sv2::BufferPool`), providing more efficient memory usage,
+//!   particularly in high-throughput scenarios.
+//! - If this feature is not enabled, a system memory buffer (`binary_sv2::BufferFromSystemMemory`)
+//!   is used for simpler applications where memory efficiency is less critical.
```

Comment:
the module is private, which means this header is for maintainer eyes only

whenever we make the module public we can change this, but for now let's document things as they are
```suggestion
// # Decoder
//
// Provides utilities for decoding messages held by Sv2 frames, with or without Noise protocol
// support.
//
// It includes primitives to both decode encoded standard Sv2 frames and to decrypt
// and decode Noise-encrypted encoded Sv2 frames, ensuring secure communication when required.
//
// ## Usage
// All messages passed between Sv2 roles are encoded as Sv2 frames. These frames are decoded using
// primitives in this module. There are two types of decoders for reading these frames: one for
// regular Sv2 frames (`StandardDecoder`), and another for Noise-encrypted frames
// (`StandardNoiseDecoder`). Both decoders manage the deserialization of incoming data and, when
// applicable, the decryption of the data upon receiving the transmitted message.
//
// ### Buffer Management
//
// The decoders rely on buffers to hold intermediate data during the decoding process.
//
// - When the `with_buffer_pool` feature is enabled, the internal `Buffer` type is backed by a
//   pool-allocated buffer (`binary_sv2::BufferPool`), providing more efficient memory usage,
//   particularly in high-throughput scenarios.
// - If this feature is not enabled, a system memory buffer (`binary_sv2::BufferFromSystemMemory`)
//   is used for simpler applications where memory efficiency is less critical.
```